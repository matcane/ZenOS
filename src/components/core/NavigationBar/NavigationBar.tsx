import Feather from "@expo/vector-icons/Feather";
import { useSegments } from "expo-router";
import { Pressable, View } from "react-native";

import { useTheme } from "@/hooks/core";
import { useNav } from "@/hooks/core/useNav";
import { baseStyle } from "@/styles/baseStyle";
import { coreStyles } from "@/styles/core";
import { Colors } from "@/theme";

const { navigationBarContainer, backNavButton } = coreStyles;

export default function NavigationBar() {
  const theme = useTheme();
  const segments = useSegments();
  const { isRoot, navBarColor, homeNavigation, backNavigation } = useNav();

  const iconSize = baseStyle.sizeSM.height;
  const backgroundColor = isRoot ? undefined : { backgroundColor: navBarColor };
  const navBarButtonHitSlop = {
    bottom: 15,
    left: 20,
    right: 20,
    top: 15,
  };
  const color = isRoot ? Colors.dark.text : theme.text;
  const canBack = !isRoot;

  if ((segments.length === 1 && segments[0] === "sign-in") || segments[0] === "sign-up")
    return null;

  return (
    <View style={[navigationBarContainer, backgroundColor]}>
      <Feather name="menu" size={iconSize} color={color} />

      <Pressable hitSlop={navBarButtonHitSlop} disabled={!canBack} onPress={homeNavigation}>
        <Feather name="square" size={iconSize} color={color} />
      </Pressable>

      <Pressable hitSlop={navBarButtonHitSlop} disabled={!canBack} onPress={backNavigation}>
        <Feather name="triangle" style={backNavButton} size={iconSize} color={color} />
      </Pressable>
    </View>
  );
}
