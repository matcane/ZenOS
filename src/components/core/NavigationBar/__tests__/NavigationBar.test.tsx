import { render } from "@testing-library/react-native";

import NavigationBar from "../NavigationBar";

jest.mock("expo-router", () => ({
  useSegments: jest.fn(() => []),
}));

describe("NavigationBar component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const tree = render(<NavigationBar />);
    expect(tree).toMatchSnapshot();
  });
});
