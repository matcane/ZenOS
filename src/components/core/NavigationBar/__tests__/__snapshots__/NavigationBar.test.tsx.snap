// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 20,
        "height": 45,
        "justifyContent": "space-evenly",
      },
      undefined,
    ]
  }
>
  <Text />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      {
        "bottom": 15,
        "left": 20,
        "right": 20,
        "top": 15,
      }
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      {
        "bottom": 15,
        "left": 20,
        "right": 20,
        "top": 15,
      }
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text />
  </View>
</View>
`;
